### PR TITLE
Manage error present in external service response.

### PR DESCRIPTION
**Pull request by kadhikari**
Fix applied to api /freefloatings_nearby and /stop_schedules(vehicle_occupancies) 
For details : https://jira.kisio.org/browse/NAVITIAII-3475

### CODE REVIEW

File: source/jormungandr/jormungandr/external_services/external_service.py [hunk 1]
```diff
@@ -89,10 +85,10 @@ def get_response(self, arguments):
 
     @classmethod
     def _check_response(cls, response):
-        if not response:
+        if response is None:
             raise ExternalServiceError('impossible to access external service')
         if response.status_code == 503:
-            raise ExternalServiceUnavailable('forseti responded with 503')
+            raise ExternalServiceError('forseti responded with 503')
```
Comment (azime) [inline source/jormungandr/jormungandr/external_services/external_service.py]: Can you add the message returned by Forseti, No?

image

File: source/jormungandr/tests/vehicle_occupancy_tests.py [hunk 2]
```diff
@@ -151,3 +153,12 @@ def test_occupancy_empty_list_occupancies(self):
         stop_schedules = response['stop_schedules'][0]['date_times']
         assert len(stop_schedules) == 1
         assert "occupancy" not in stop_schedules[0]
+
+    def test_occupancy_empty_list_occupancies_on_error(self):
+        query = self.query_template_scs.format(
+            sp='stopQ2', dt='20160103T100000', data_freshness='&data_freshness=realtime'
+        )
+        response = self.query_region(query)
+        stop_schedules = response['stop_schedules'][0]['date_times']
```
Comment (azime) [inline source/jormungandr/tests/vehicle_occupancy_tests.py]: Minor:
```suggestion
        date_times = response['stop_schedules'][0]['date_times']
```

File: source/jormungandr/tests/vehicle_occupancy_tests.py [hunk 3]
```diff
@@ -102,6 +102,8 @@ def _call_webservice(self, arguments):
             }
         if stop_id == "stopP2" and vehicle_journey_id == 'vehicle_journey:vjP:1:modified:0:bib':
             json = {"vehicle_occupancies": []}
+        if stop_id == "stopQ2":
+            return MagicMock(return_value=None)
```
Comment (azime) [inline source/jormungandr/tests/vehicle_occupancy_tests.py:106]: Can you test the forseti 503 return code?

File: source/jormungandr/jormungandr/external_services/external_service.py (hunk 1 above)
Comment (kadhikari) [reply source/jormungandr/jormungandr/external_services/external_service.py]: Infact we dont 't need this exception. It is already handled in the following lines.

File: source/jormungandr/tests/vehicle_occupancy_tests.py (hunk 2 above)
Comment (kadhikari) [reply source/jormungandr/tests/vehicle_occupancy_tests.py]: :+1:

File: source/jormungandr/tests/vehicle_occupancy_tests.py (hunk 3 above)
Comment (kadhikari) [reply source/jormungandr/tests/vehicle_occupancy_tests.py:106]: I tried to test with status_code 503.

**Review by azime — APPROVED** [review]

**Review by pbench — APPROVED** [review]

**Pull request merged**